tools/dissector/sample_dissector.cpp

```diff
@@ -196,37 +193,94 @@ namespace OpenDDS
       return ns_stack_.back();
     }
 
-    void Sample_Base::add_protocol_field(enum ftenum ft, field_display_e fd) {
-      Field_Context * fc = get_context();
-      Sample_Manager::instance().add_protocol_field(
-        &fc->hf_, get_ns(), fc->label_, ft, fd
-      );
+    void Sample_Base::push_idl_name(std::string name) {
+      Sample_Base::clear_ns();
+
+      // If name begins with "IDL:", remove it
+      const std::string idl_prefix("IDL:");
+      if (name.substr(0, idl_prefix.size()) == idl_prefix) {
+          name.erase(0, idl_prefix.size());
+      }
+
+      // If name contains ':', remove everything after the last ':'
+      size_t l = name.rfind(":");
+      if (l != std::string::npos) {
+          name.erase(l, name.size() - l);
+      }
+
+      // Push namespace when we find a '/'
+      name.push_back('/'); // For last namespace
+      l = 0;
+      for (size_t index = 0; index != name.size(); index++) {
+          if (name[index] == '/') {
+              Sample_Base::push_ns(name.substr(index - l, l));
+              l = 0;
+          } else {
+              l++;
+          }
+      }
+    }
+
+    void Sample_Base::add_protocol_field() {
+      Field_Context* fc = get_context();
+      if (fc) {
+        Sample_Manager::instance().add_protocol_field(fc->hf_info_);
+      }
     }
 
-    Sample_Field::Sample_Field (IDLTypeID id, const std::string &label)
-      : label_ (label),
-        type_id_ (id),
-        nested_ (0),
+    Field_Context* Sample_Base::create_context(ftenum ft, field_display_e fd) {
+      if (field_contexts_.count(get_ns())) {
+        throw Sample_Dissector_Error(
+          std::string("Could not create context becuase it already exists for: ") +
+          get_ns()
+        );
+      }
+
+      Field_Context* fc = new Field_Context;
+      fc->long_name_ = get_ns();
+      fc->short_name_ = get_label();
+      fc->hf_ = -1;
+      fc->hf_info_ = { &fc->hf_, {
```

Comment:
The syntax of `x = {a, b};` is only in C++11 (unless I'm missing something here?), we should use something that's backwards compatible OR document the requirements on compilers and flags (for example, GCC 5.x and earlier requires -std=c++11).

Reply:
[cppreference.com says that this works for pre C++11 if the class/struct has no constructor](https://en.cppreference.com/w/cpp/language/aggregate_initialization) and I just tested it and the syntax works with `g++ -std=c++03`.

Reply:
That's for initialization, not for assignment.  GCC should at least be warning for this.

Reply:
You are right, I was thinking it would work for everywhere if it worked for initialization. Would it be alright if I put it inside of constructor member initialization list? I could also create a default value variable and copy it.

Reply:
Yes, Field_Context can have whatever constructor(s) you find useful.

Reply:
So GCC gives the same warning for struct initializer list in a member initializer list as it does for trying to assign it (the Extended Initializer Warning). It's not really a problem, but it wasn't what I was expecting would happen since I would have thought using member initializer lists would be equivalent to assigning a new variable.